s3/src/main/scala/akka/stream/alpakka/s3/scaladsl/S3Client.scala

```diff
@@ -48,10 +45,13 @@ object S3Client {
   def apply()(implicit system: ActorSystem, mat: Materializer): S3Client =
     new S3Client(S3Settings(system.settings.config))
 
-  def apply(credentials: AWSCredentials, region: String)(implicit system: ActorSystem, mat: Materializer): S3Client = {
-    val settings = S3Settings
-      .apply(system.settings.config)
-      .copy(awsCredentials = credentials, s3Region = region)
+  def apply(credentials: AWSCredentialsProvider, region: String)(implicit system: ActorSystem,
```

Comment:
Could we keep the old signature as well, and just call this method wrapping the AWSCredentials with a static provider, that way we avoid breaking everyones code.

Reply:
Will do. However, the code relying on config will still break, do you want me to do a compatibility layer for that?